src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/StartToolStep.groovy

```diff
@@ -283,34 +243,41 @@ class StartToolStep extends Step {
 
             Process process = new ProcessBuilder().command(args.toCommandArray()).start()
 
-            boolean isStarted = false
-            long endTimeMillis = System.currentTimeMillis() + (long) timeout * 1000L
-            while (timeout <= 0 || System.currentTimeMillis() < endTimeMillis) {
-                if ((isStarted = process.isAlive())) {
-                    break
-                } else {
-                    Thread.sleep(1000)
+            boolean  isConnected = checkToolConnection()
+            int exitCode = 0
+            if (!isConnected) {
+                try  {
+                    exitCode = process.exitValue()
+                }  catch (IllegalThreadStateException ignore){
+                    process.destroy()
+                    throw new AbortException(
+                            "Timeout of ${this.timeout} seconds exceeded for connecting to ${toolName}! " +
+                                    "Please ensure the tool is correctly configured and consider restarting it.")
                 }
+            } else {
+                return
             }
 
-            if (!isStarted) {
-                throw new AbortException(
-                        "Timeout of ${this.timeout} seconds exceeded for starting ${toolName}! " +
-                        "Please ensure that the tool is correctly configured and accessible.")
+            if (exitCode == 99) {
+                throw new AbortException("No valid license found for ${toolName}! " +
+                        "Please ensure the license is not expired or corrupted.")
             }
+
+            throw new AbortException(
+                    "${toolName} did not start correctly and stopped with exit code ${exitCode} and " +
+                            "was terminated within the timeout of ${timeout} seconds.")
```

Comment:
`and was terminated` is duplicate of ``stopped` and can be removed from the sentence